tests/formatconverter/formatconvertertest.cpp

```diff
@@ -54,155 +53,185 @@ using std::string;
 // equally, therefore it is enough to choose just one other for testing, %d
 // will do.
 
-class FormatConverterTestCase : public TestCase
+void check(const wxString& input, const wxString& expectedScanf,
+    const wxString& expectedUtf8,
+    const wxString& expectedWcharUnix,
+    const wxString& expectedWcharWindows)
 {
-    CPPUNIT_TEST_SUITE(FormatConverterTestCase);
-        CPPUNIT_TEST(format_d);
-        CPPUNIT_TEST(format_hd);
-        CPPUNIT_TEST(format_ld);
-        CPPUNIT_TEST(format_s);
-        CPPUNIT_TEST(format_hs);
-        CPPUNIT_TEST(format_ls);
-        CPPUNIT_TEST(format_c);
-        CPPUNIT_TEST(format_hc);
-        CPPUNIT_TEST(format_lc);
-        CPPUNIT_TEST(format_S);
-        CPPUNIT_TEST(format_hS);
-        CPPUNIT_TEST(format_lS);
-        CPPUNIT_TEST(format_C);
-        CPPUNIT_TEST(format_hC);
-        CPPUNIT_TEST(format_lC);
-        CPPUNIT_TEST(testLonger);
-    CPPUNIT_TEST_SUITE_END();
-
-    void format_d();
-    void format_hd();
-    void format_ld();
-    void format_s();
-    void format_hs();
-    void format_ls();
-    void format_c();
-    void format_hc();
-    void format_lc();
-
-    void format_S();
-    void format_hS();
-    void format_lS();
-    void format_C();
-    void format_hC();
-    void format_lC();
-    void testLonger();
-
-    void doTest(const char *input, const char *expectedScanf,
-                                   const char *expectedUtf8,
-                                   const char *expectedWcharUnix,
-                                   const char *expectedWcharWindows);
-    void check(const wxString& input, const wxString& expectedScanf,
-                                      const wxString& expectedUtf8,
-                                      const wxString& expectedWcharUnix,
-                                      const wxString& expectedWcharWindows);
-};
-
-void FormatConverterTestCase::format_d()
+    // all of them are unused in some build configurations
+    wxUnusedVar(expectedScanf);
+    wxUnusedVar(expectedUtf8);
+    wxUnusedVar(expectedWcharUnix);
+    wxUnusedVar(expectedWcharWindows);
```

Comment:
It could be useful to use `INFO()` to log all of those: this will give useful information in case of a test failure and will avoid the need for these lines.